kindle.js

```diff
@@ -59,12 +59,28 @@ class Kindle {
 			return "only support 'author' or 'title' criteria";
 		}
 	}
+	search(keywords) {
+		const typedKeyword = keywords.toLowerCase().trim();
+		const result = [];
+		this._library.filter(function(eBook) {
+			if (
+				eBook.title.toLowerCase().includes(typedKeyword) ||
+				eBook.author.toLowerCase().includes(typedKeyword)
+			) {
+				result.push(eBook);
+			}
```

Comment:
**filter retorna un nuevo array**, que es el que deberias usar en result, no es necesario el if y el push y todo eso, `const result = this.library.filter (fn)` seria la idea, igualmente tu funcion filtradora creo que no funciona, estas usando `typedKeyword` en otro scope, o sea me refiero a que deberas pasarlo como parametro dentro de la funcion filtradora, lo probaste?